### DISCUSSION TITLE
Support incremental builds

### DISCUSSION THREAD

**Discussion by kawpii** [vercel/next.js#30716] [Ideas]
### Describe the feature you'd like to request

As I understood the way Next ISR do not have anything similar to [Gatsby's incremental](https://www.gatsbyjs.com/blog/2020-04-22-announcing-incremental-builds/) builds which only build pages which were changed on the 2nd run, what Next have is persistent cache which users can roleback. (no sharing of cache between deploys, so second build still build the unchanged pages). Am I right here?

But as I read here from a Vercel user [here](https://github.com/vercel/next.js/discussions/30244#discussion-3647160) Vercel platform support persistent cache between deploys, which seems to be inherent to the Vercel platform.

### Describe the solution you'd like

Support incremental builds similar to Gatsby, this would not require data layer diffing like Gatsby just hashing built files on the file system and comparing would do.

### Describe alternatives you've considered

Using Gatsby for JAMstack sites or using Vercel platform.

**Comment by kawpii** [comment]
@timneutkens Why close the issue? Is not it a focus in Nextjs?

**Reply by timneutkens** [reply]
It's not closed. It's converted to discussions as per the feature request template

**Reply by jadamconnor** [reply]
It would be nice for large sites to be able to make minor code changes and build only those changes.
